File: Runtime/StateAttributeValue.vb
'*********************************************************************************************
' ST-Sim: A SyncroSim Module for the ST-Sim State-and-Transition Model.
'
' Copyright © 2007-2017 Apex Resource Management Solution Ltd. (ApexRMS). All rights reserved.
'
'*********************************************************************************************

''' <summary>
''' State attribute value class
''' </summary>
''' <remarks></remarks>
Friend Class StateAttributeValue
    Inherits AttributeValueBase

    Public Sub New(
        ByVal stateAttributeTypeId As Integer,
        ByVal stratumId As Nullable(Of Integer),
        ByVal secondaryStratumId As Nullable(Of Integer),
        ByVal iteration As Nullable(Of Integer),
        ByVal timestep As Nullable(Of Integer),
        ByVal stateClassId As Nullable(Of Integer),
        ByVal minimumAge As Nullable(Of Integer),
        ByVal maximumAge As Nullable(Of Integer),
        ByVal value As Double)

        MyBase.New(
            stateAttributeTypeId,
            stratumId,
            secondaryStratumId,
            iteration,
            timestep,
            StateClassId,
            MinimumAge,
            MaximumAge,
            Value)

    End Sub

End Class

File: Runtime/StateAttributeValue.cs
// ST-Sim: A SyncroSim Module for the ST-Sim State-and-Transition Model.
// Copyright © 2007-2018 Apex Resource Management Solution Ltd. (ApexRMS). All rights reserved.

namespace SyncroSim.STSim
{
    internal class StateAttributeValue : AttributeValueBase
    {
        public StateAttributeValue(
            int stateAttributeTypeId, int? stratumId, int? secondaryStratumId, int? tertiaryStratumId, 
            int? iteration, int? timestep, int? stateClassId, int? minimumAge, int? maximumAge, double value) : 
            base(stateAttributeTypeId, stratumId, secondaryStratumId, tertiaryStratumId, iteration, 
                timestep, stateClassId, minimumAge, maximumAge, value)
        {
        }
    }
}
